Created Publishers for added external sensors: line and camera

blacktorch opened the pull request.

blacktorch committed:
Created Publishers for added external sensors: line and camera

pgavigan commented:
Looks good. My only real comment for now would be to suggest adding some more comments explaining what your functions do. For example: "Line sensor publishes the .... on the .... topic. Data is received from the ... sensor connected to pins ..."

pgavigan reviewed (approved):
Can merge, but suggest adding more comments to explain functionality of each file/ function.

blacktorch committed:
added driver test for roomba

blacktorch committed:
added driver test for roomba

blacktorch committed:
added driver test for roomba

blacktorch committed:
made executable

blacktorch committed:
Merge branch 'external-sensors' of https://github.com/NMAI-lab/saviRoomba into external-sensors

blacktorch committed:
added action decoder

blacktorch committed:
updated asl

blacktorch committed:
Edited configuration files

blacktorch committed:
updated action decoder

blacktorch committed:
updated action decoder

blacktorch committed:
added test drive script

blacktorch committed:
made testDrive.py executeable

blacktorch committed:
remote controller added

blacktorch committed:
added test drive script

blacktorch committed:
Updated ASL plans

blacktorch committed:
updated settings

blacktorch committed:
updated plans

blacktorch committed:
updated plans

blacktorch committed:
change debug info

blacktorch committed:
adjusted line sensing

blacktorch committed:
adjusted camera settings

blacktorch committed:
updated QR reader

blacktorch committed:
updated QR reader

blacktorch committed:
updated ASL plan

blacktorch committed:
updated ASL plan

blacktorch committed:
updated line sensor

blacktorch committed:
added app receiver

blacktorch committed:
added app receiver

blacktorch committed:
made app_receiver executable

blacktorch committed:
added app receiver

blacktorch committed:
added app receiver

blacktorch committed:
added battery manager

blacktorch committed:
added battery manager and docker

pgavigan committed:
First iteration of new ASL

Navigating ASL program. This is a first step. There WILL be bugs.

pgavigan committed:
Using unification to combine line following plans

pgavigan committed:
Added map specific navigation rules.

pgavigan committed:
Added map image for reference

pgavigan committed:
Navigation rules use word destination, not dest

pgavigan committed:
Applied navigation rules to plans

blacktorch committed:
Merge pull request #6 from NMAI-lab/nav

Navigation ASL

blacktorch committed:
updated asl and implemented turn action

pgavigan committed:
Did some debugging in ASL based on documenting for the paper

I think I found some bugs when documenting the ASL for the paper. Suggested fixes are here.

pgavigan committed:
Another possible bug fix

Need to ensure the only way out of delivering mail is actually delivering mail

pgavigan committed:
I suspect that these should not be there.

pgavigan committed:
Removed recursion that I don't think should be there.

pgavigan committed:
Reworked implementation of docking the robot

blacktorch committed:
Merge pull request #7 from NMAI-lab/aslDebug

Did some debugging in ASL based on documenting for the paper

pgavigan committed:
Merge branch 'master' into external-sensors
